src/cursor/cur_prepared_discover.c

```diff
@@ -81,14 +81,20 @@ __cursor_prepared_discover_close(WT_CURSOR *cursor)
     WT_TXN_GLOBAL *txn_global;
     WT_TXN_OP *op;
     uint64_t i, j;
+    uint64_t unclaimed_count;
     cursor_prepare = (WT_CURSOR_PREPARE_DISCOVERED *)cursor;
     CURSOR_API_CALL_PREPARE_ALLOWED(cursor, session, close, NULL);
 
     txn_global = &S2C(session)->txn_global;
     pending_prepare_items = &txn_global->pending_prepare_items;
+    unclaimed_count = 0;
     if (pending_prepare_items->hash != NULL) {
         for (i = 0; i < pending_prepare_items->hash_size; i++) {
             while ((item = TAILQ_FIRST(&pending_prepare_items->hash[i])) != NULL) {
+                if (!item->claimed) {
```

Comment:
We can exit early here when we find the first unclaimed prepare id, but it will cause memory leak since we still need to clean up the hash map. I'm open to suggestion how to do this